### DISCUSSION TITLE
WinUAE Permedia 2 (2D-only) and NCR 77C22E+/77C32BLT emulation

### DISCUSSION THREAD

**Discussion by tonioni** [86Box/86Box#4282] [Ideas]
Hi,

I wrote Permedia 2 and NCR SVGA chip emulation for WinUAE Amiga emulator (There are Amiga graphics boards based on both chips). Note that it is based on mix of old 86box and pcem "framework", not current version. I plan to do full 86box merge someday but it currently lacks features that don't matter in PC world (see below).

Permedia 2 emulation is 2D-only (including all 2D blitter features that Amiga drivers use), 3D does not interest me. No (S)VGA mode included because I have no way to test it. NCR chip is fully emulated. Both are included in public winuae source tree.

Maybe this is useful for 86box too? I have already "borrowed" other (S)VGA chip emulators from 86box and pcem. Maybe even someone is interested in adding 3D parts?

I also have small request: is there possibility to include official big-endian mode ((S)VGA chips that have it switchable) support, even if it is inside #ifdefs (I understand it would be waste of time in PC-only emulator where big-endian is never used) and also have some official framework for vblank interrupts because almost all Amiga drivers expect working (S)VGA vblank interrupt. There is also some other minor changes that I have added (panning was wrong on some chips, big-endian modes in svga_render etc)

**Comment by TC1995** [comment]
I'm interested in the NCR SVGA one. :)
